### PR TITLE
Homepage Featured Section Implementation

### PR DESCRIPTION
**Pull request by vishalbandre**
### Homepage Featured Items Section:

N.B. Homepage Section for PopSugar.com

Drupal Admin Efforts:
    - A view that fetches 4 data items from different categories.
    - Each item is made up of featured image/thumbnail, title and tags involved.
    - Newly, created view has been added to content block on homepage.

Code Changes:
  - Homepage sections specific css will be served with dedicated asset library. Declared at web/themes/custom/pop/pop.libraries.yml and attached to front page template with attach_library twig function. It can be found here: web/themes/custom/pop/templates/page--front.html.twig
  - Homepage specific sass files are arranged in homepage directory at web/themes/custom/pop/sass/layout/.

You can refer related issue here: [Issue](https://github.com/vishalbandre/ps-clone/issues/7)

Screenshorts:
Desktop View:
Screenshot from 2022-02-08 15-17-26

Mobile View:
Screenshot from 2022-02-08 15-09-02

### CODE REVIEW

File: web/modules/custom/pop/pop.info.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+name: Pop Essentials
+description: Provides Pop theme essentials.
+type: module
+core_version_requirement: ^8 || ^9
+package: Custom
```
Comment (JayKandari) [inline web/modules/custom/pop/pop.info.yml:5]: `No new lines at the end of file`.

File: .gitignore [hunk 2]
```diff
@@ -52,4 +52,5 @@ web/profiles/contrib/
 web/libraries/
 
 # Theme
-web/themes/custom/pop/css/*
\ No newline at end of file
+web/themes/custom/pop/css/*
+web/themes/custom/pop/css
```
Comment (JayKandari) [inline .gitignore:56]: `No new lines at the end of file`.

File: web/themes/custom/pop/pop.info.yml [hunk 3]
```diff
@@ -8,6 +8,7 @@ libraries:
   - pop/global-styling
   # - pop/animate
   - pop/custom
+  # - pop/homepage
```
Comment (JayKandari) [inline web/themes/custom/pop/pop.info.yml:11]: Remove  unwanted comment.

**Review by JayKandari — CHANGES_REQUESTED** [review]

File: web/modules/custom/pop/pop.info.yml (hunk 1 above)
Comment (vishalbandre) [reply web/modules/custom/pop/pop.info.yml]: Made suggested changes.

File: .gitignore (hunk 2 above)
Comment (vishalbandre) [reply .gitignore]: Made suggested changes.

File: web/themes/custom/pop/pop.info.yml (hunk 3 above)
Comment (vishalbandre) [reply web/themes/custom/pop/pop.info.yml]: Made suggested changes. Removed unwanted comments.

File: web/themes/custom/pop/sass/layout/homepage/_container.scss [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    }
+    .row {
+        margin: 0;
+        padding: 0;
+        .col {
+            .region,
+            .region-content {}
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (vishalbandre) [reply web/themes/custom/pop/sass/layout/homepage/_container.scss:14]: Made suggested changes.

File: web/themes/custom/pop/pop.theme [hunk 5]
```diff
@@ -163,4 +163,20 @@ function pop_form_alter(&$form, $form_state) {
...
+ * Implements hook_views_query_alter().
+ */
+function pop_views_query_alter(ViewExecutable $view, QueryPluginBase $query) {
+
+  // homepage_featured view
+  if ($view->id() == 'homepage_featured') {
+    echo 'View accessed.';
+  }
+
+  echo 'View accessed.';
```
Comment (JayKandari) [inline web/themes/custom/pop/pop.theme:181]: Pass the user facign text inside a `t()` function. Eg: `t('View accessed');`

Change the same at other places as well.

Comment (vishalbandre) [reply web/themes/custom/pop/pop.theme]: Made suggested changes. Thanks.

**Pull request merged**